server/service/apple_mdm.go

```diff
@@ -3171,15 +3166,30 @@ func (svc *MDMAppleDDMService) handleDeclarationsResponse(ctx context.Context, e
 	}
 	level.Debug(svc.logger).Log("msg", "parsed declarations request", "type", parts[1], "identifier", parts[2])
 
-	// TODO: Validate declarationType?
-	d, err := svc.ds.MDMAppleDDMDeclarationsResponse(ctx, fleet.MDMAppleDeclarationType("com.apple."+parts[1]), parts[2], hostUUID)
+	d, err := svc.ds.MDMAppleDDMDeclarationsResponse(
+		ctx, fleet.MDMAppleDeclarationCategory("com.apple."+parts[1]),
+		parts[2],
+		hostUUID,
+	)
 	if err != nil {
 		return nil, ctxerr.Wrap(ctx, err, "getting declaration")
 	}
-	b, err := json.Marshal(d)
+
+	// unmarshall into a temporary map in order to add the token.
+	// we do this at this stage because tokens are purely managed by Fleet,
+	// and we don't want to store a modified version of what's provided by
+	// the IT admin.
+	//
+	// This mimics what we do for CommandUUID, but can be revisited.
```

Comment:
I thought Fleet would be adding token to the declaration when it is stored in the DB, which would save us the unmarshal/marshal cycles when we serve it to the client. Is there a reason not to do it on the upload?

Reply:
yes, when the IT admin downloads the declaration we don't want to include anything else.

in general keeping whatever the IT admin uploaded as-is has been our direction for profiles (windows and mac) so I think just for consistency we should do the same (at the cost of a few memory cycles)

we can re-evaluate of course!

Reply:
Sounds good! Apologies for misunderstanding the intent there.

Reply:
nah, I think this was not properly specified, sorry about that!